### PR TITLE
Add Hive target max file size

### PR DESCRIPTION
**Pull request by dain**
Hive page sink rolls the writer when the writer reports a size
larger then the target maximum file size.  This can be configured with
as session or configuration property. The default is 1GB.

### CODE REVIEW

**Comment by joshthoward** [comment]
Requesting @losipiuk to review since we have users requesting this and it should go quick.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java [hunk 1]
```diff
@@ -1618,6 +1620,40 @@ public void testCreateTableWithUnsupportedType()
...
+        assertThat(computeActual(selectFileInfo).getRowCount()).isEqualTo(3);
+        assertUpdate("DROP TABLE test_max_file_size");
+
+        // Write table with small limit and verify we get multiple files per node near the expected size
+        // Writer writes chunks of rows that are about 40k
+        DataSize maxSize = DataSize.of(40, Unit.KILOBYTE);
+        session = Session.builder(getSession())
+                .setSystemProperty("task_writer_count", "1")
+                .setCatalogSessionProperty("hive", "target_max_file_size", maxSize.toString())
+                .setCatalogSessionProperty("hive", "hive_storage_format", "TEXTFILE")
```
Comment (losipiuk) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java:1643]: please add a comment why `TEXTFILE` is important here (or drop if it is not). Also set same `hive_storage_format` here and in session used for sanity check above.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java [hunk 2]
```diff
@@ -145,6 +146,7 @@
     private boolean optimizeSymlinkListing = true;
 
     private boolean legacyHiveViewTranslation;
+    private DataSize targetMaxFileSize = DataSize.of(1, GIGABYTE);
```
Comment (losipiuk) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:149]: I am a bit curious where did the default value come from. But 1GB seems a safe default.

**Review by losipiuk — APPROVED** [review]
LGTM. One suggestion in test.

**Comment by alec-heif** [comment]
Just wanted to chime in that my org would also like this PR to land, and it seems like there aren't any major blockers - can this be landed soon? Thank you!

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java (hunk 2 above)
Comment (dain) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java:149]: It is arbitrary.  It makes for pretty large files by default... generally it is large enough that most things aren't going to hit it.  We can easily change this in the future.

**Pull request merged**